contracts/meta-aggregator/MetaAggregatorSwapContract.sol

```diff
@@ -120,23 +150,68 @@ contract MetaAggregatorSwapContract is
             receiver,
             isDelegate
         );
+        emit ETHSwappedForToken(amountOut, address(tokenOut), receiver);
+    }
+
+    /**
+     * @dev Swaps ETH using delegatecall.
+     * This function is called when the swap needs to happen in a delegatecall context.
+     * @param tokenIn must be the native token.
+     * @param tokenOut The ERC20 token to swap to.
+     * @param aggregator The address of the aggregator to use for the swap.
+     * @param swapData The data required for the swap.
+     * @param amountIn The amount of ETH to swap.
+     * @param minAmountOut The minimum amount of tokenOut expected.
+     * @param receiver The address to receive the tokenOut.
+     * @param isDelegate Indicates if the swap is in a delegatecall context.
```

Comment:
Any specific reason you added the parameter `isDelegate`? The function description says "Swaps ETH using delegatecall."

Reply:
It is for the aggregator called by our contract. If enso this is true or others this is false.